lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -219,8 +219,17 @@ struct ObjectModelIR {
     auto memPaths = instancePathCache.getAbsolutePaths(mem);
     SmallVector<Value> memoryHierPaths;
     SmallVector<Value> finalInstanceNames;
-    // Memory hierarchy is relevant only for memories under DUT.
-    bool inDut = instanceInfo.anyInstanceUnderEffectiveDut(mem);
+    // Add metadata for memory paths that are in the design.  There are two
+    // slightly different code paths here.  If no instance is in the design,
+    // then just skip the memory.  If some instances are in the designs and some
+    // are not, then add paths for memories in the design.  For memories which
+    // are then _not_ in the design, give them unresolvable distinct attribute
+    // paths.  The LowerClasses pass will later treat these as "optimized away"
+    // and create an empty path.
+    //
```

Comment:
Didn't realize this behavior existed. As far as I can tell, any downstream tool parsing the OMIR would reject such an optimized away empty path, so maybe its safe to assume this case is not executed in any real world design. Thanks for documenting it, in a followup PR I can try to drop such a memory from the metadata.